frontend/src/hooks/useChangeRequestsEnabled.ts

```diff
@@ -1,10 +1,39 @@
+import React from 'react';
 import useUiConfig from './api/getters/useUiConfig/useUiConfig';
+import { useChangeRequestConfig } from './api/getters/useChangeRequestConfig/useChangeRequestConfig';
 
-export const useChangeRequestsEnabled = (environment?: string) => {
-    // it can be swapped with proper settings instead of feature flag
+export const useChangeRequestsEnabled = (projectId: string) => {
     const { uiConfig } = useUiConfig();
-    return (
-        Boolean(uiConfig?.flags?.changeRequests) &&
-        (environment === 'production' || typeof environment === 'undefined')
+    const { data } = useChangeRequestConfig(projectId);
+
+    const isChangeRequestEnabled = React.useCallback((): boolean => {
+        return Boolean(uiConfig?.flags.changeRequests);
+    }, [uiConfig?.flags.changeRequests]);
+
+    const isChangeRequestConfigured = React.useCallback(
+        (environment: string): boolean => {
+            const enabled = data.some(draft => {
+                return (
+                    draft.environment === environment &&
+                    draft.changeRequestEnabled
+                );
+            });
+
+            return isChangeRequestEnabled() && enabled;
+        },
+        [data]
     );
+
+    const isBannerEnabled = React.useCallback((): boolean => {
```

Comment:
banner impl leaking out here. I'd find a more generic purpose of this flag. Sth like isChangeRequestEnabledInAnyEnv?